mps2_an521: add a new board target variant: mps2_an521_remote for the dual core samples and applications

ioannisg opened the pull request:
Fixes #34570 

Background:
- mps2_an521 shared the same build target: `mps2_an521_nonsecure` for
   - builds for the non-secure domain
   - builds for the remote core (in dual image applications)
 - this created a problem with openamp samples running on the remote core, with the `_TRUSTED_EXECUTION_NONSECURE=y `  (while essentially they are secure only builds)

1. We fix this by introducing a new target variant, `mps2_an521_remote`, that should be exclusively used in dual core samples, for the remote core. We set the testing flag of this boards to false, so it does not "interfere" with CI. Note that this target is only used in dual core samples. We do not enable QEMU testing on it.

2. We clean up the workarounds in the board code, to mutually exclude OPENAMP and TFM, as non-needed any more.

ioannisg committed:
boards: mps2_an521: add an mps2_an521_remote target

Add a new target for MPS2 AN521, called mps2_an521_remote.
This target is used in dual image MPS2 AN521 builds, for
the 'remote' target. It is indentical to mps2_an521_nonsecure,
except for the fact that it does not enable TrustZone-M and
it does not set the security domain to Non-Secure.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
boards: mps2_an521: remove dependencies for BUILD_WITH_TFM

We do not need to have dependencies any more on !OPENAMP
for BUILD_WITH_TFM in the mps2 an521 target, because we
now have different targets for the non-secure version of
the board and the remote core.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
samples: openamp: use mps2_an521_remote target for remote core

When building and running the openamp sample on MPS2 AN521,
use the mps2_an521_remote target to build the Zephyr image
for the remote core.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

microbuilder reviewed (approved):
Tested via:

```bash
$ west build -p auto -b mps2_an521 sample
$ cd build
$ ninja run
```

There seems to be an issue with `$ west build -p auto -b mps2_an521 zephyr/samples/subsys/ipc/openamp/ -t run` but that's out of scope with this PR, and can be addressed elsewhere.

oyvindronningstad reviewed (approved):
LGTM!

ioannisg committed:
samples: rpmsg_service: use mps2_an521_remote target for remote core

When building and running the rpmsg_service sample on MPS2 AN521
use the mps2_an521_remote target to build the Zephyr image
for the remote core.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
> There seems to be an issue with $ west build -p auto -b mps2_an521 zephyr/samples/subsys/ipc/openamp/ -t run but that's out of scope with this PR, and can be addressed elsewhere.

...As long as it does not break CI here :) i should be un related to this PR, I agree
